client/src/components/survey-view/checkbox-button/CheckboxButton.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import SingleCheckboxButton from './SingleCheckboxButton';
+
+class CheckboxButton extends React.Component {
```

Comment:
Could you make it a functional component use `useState` ?

Reply:
like with hooks?

Reply:
I can, but is this necessary?

Reply:
Using functional components is typically best practices now with React - should be used as much as possible 

https://www.codeinwp.com/blog/react-best-practices/

Reply:
useCallback would be useful here.

Reply:
i used useReducer, that way I don't need useCallback